### PR TITLE
fix(migrate): alter column `value_json` size in `regulation_riskdetect_rule` table

### LINKED ISSUE #3278: [Bug]: Error reporting when creating too many risk identification rules
### ODC version

ODC 4.3.1

### OB version

No related.

### What happened?

Error reporting when creating too many risk identification rules.Error accessing the metadata database.
image

### What did you expect to happen?

No error reporting.

### How can we reproduce it (as minimally and precisely as possible)?

Create a lot of risk identification.

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by yiminpeng**
#### What type of PR is this?

fix

#### What this PR does / why we need it:

back ground: At present, a maximum of 20 can be created, if it exceeds reporting: Data truncation: (conn=3221624529) Data too long for column 'value_json' at row 1. The expected number of rules that can be built is 100.

root cause: The data table field exceeds the limit.

what we do: Alter column `value_json` size from varchar(2048) to medium text in `regulation_riskdetect_rule` table

#### Which issue(s) this PR fixes:

Fixes #3278 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_4_alter_regulation_riskdetect_rule.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--
+-- Alter column `value_json` size from varchar(2048) to varchar(10240) in `regulation_riskdetect_rule` table
+--
+alter table `regulation_riskdetect_rule` modify column `value_json` varchar(10240) default null comment 'json of rule value';
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_4_alter_regulation_riskdetect_rule.sql:4]: 4.3.2.4 is occupied, 4.3.2.11 is available

Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_4_alter_regulation_riskdetect_rule.sql:4]: prefer medium text

Comment (yiminpeng) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_4_alter_regulation_riskdetect_rule.sql:4]: got it

Comment (yiminpeng) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_4_alter_regulation_riskdetect_rule.sql:4]: of course, it is also a choice. changed.

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+--
+-- Alter column `value_json` size from varchar(2048) to varchar(10240) in `regulation_riskdetect_rule` table
+--
+alter table `regulation_riskdetect_rule` modify column `value_json` mediumtext default null COMMENT 'json of rule value';
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql:4]: it is not compatible with OB227x ?
may use CHANGE COLUMN + add COLUMN replacement,

refer V_3_2_2_1__add_and_rename_task_info_column.sql for details

Comment (yiminpeng) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql:4]: got it, job finished.

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+--
+-- Alter column `value_json` size from varchar(2048) to MEDIUMTEXT in `regulation_riskdetect_rule` table
+--
+ALTER TABLE `regulation_riskdetect_rule` CHANGE COLUMN `value_json` `value_json_old` VARCHAR(2048);
+ALTER TABLE `regulation_riskdetect_rule` ADD COLUMN `value_json` MEDIUMTEXT DEFAULT NULL;
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql:5]: how to process the data that contained by `value_json_old`?

Comment (yiminpeng) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_11_alter_regulation_riskdetect_rule.sql]: update `value_json_old` to `value_json`

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — COMMENTED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**